pkgs/agenix.nix

```diff
@@ -9,29 +9,50 @@
   substituteAll,
   ageBin ? "${rage}/bin/rage",
   shellcheck,
-}:
-stdenv.mkDerivation rec {
-  pname = "agenix";
-  version = "0.13.0";
-  src = substituteAll {
-    inherit ageBin version;
-    sedBin = "${gnused}/bin/sed";
-    nixInstantiate = "${nix}/bin/nix-instantiate";
-    mktempBin = "${mktemp}/bin/mktemp";
-    diffBin = "${diffutils}/bin/diff";
-    src = ./agenix.sh;
-  };
-  dontUnpack = true;
+}: let
+  bin = "${placeholder "out"}/bin/agenix";
+in
+  stdenv.mkDerivation rec {
+    pname = "agenix";
+    version = "0.13.0";
+    src = substituteAll {
+      inherit ageBin version;
+      sedBin = "${gnused}/bin/sed";
+      nixInstantiate = "${nix}/bin/nix-instantiate";
+      mktempBin = "${mktemp}/bin/mktemp";
+      diffBin = "${diffutils}/bin/diff";
+      src = ./agenix.sh;
+    };
+    dontUnpack = true;
 
-  doCheck = true;
-  checkInputs = [shellcheck];
-  postCheck = ''
-    shellcheck $src
-  '';
+    doInstallCheck = true;
+    installCheckInputs = [shellcheck];
+    postInstallCheck = ''
+      shellcheck ${bin}
+      ${bin} -h | grep ${version}
 
-  installPhase = ''
-    install -D $src ${placeholder "out"}/bin/agenix
-  '';
+      mkdir -p /tmp/home/.ssh
+      cp -r "${../example}" /tmp/home/secrets
+      chmod -R u+rw /tmp/home/secrets
+      export HOME=/tmp/home
+      (
+        umask u=rw,g=r,o=r
+        cp ${../example_keys/user1.pub} $HOME/.ssh/id_ed25519.pub
+        chown $UID $HOME/.ssh/id_ed25519.pub
+      )
+      (
+        umask u=rw,g=,o=
+        cp ${../example_keys/user1} $HOME/.ssh/id_ed25519
+        chown $UID $HOME/.ssh/id_ed25519
+      )
 
-  meta.description = "age-encrypted secrets for NixOS";
-}
+      cd /tmp/home/secrets
+      ${bin} -d secret1.age | grep "hello"
```

Comment:
I realize this is just a POC, but just to put it out there -- especially for something like the `-d` flag, I think we should move to something more precise than just `foo in bar` (in the old style) or `| grep foo`. For example, want to makes sure there is no leading or trailing whitespace added by an `echo` without `-n` (as opposed to `printf`) somewhere.

If sticking with bash, maybe we should write a few utility functions (that themselves should probably be tested) for this purpose, even if they're just giving a more friendly name to what is in effect `test "$(cmd)" = "foo"`.  Or maybe not, just moving to `test ... = ""` would probably be fine.

Personally, I find these kinds of tests much easier to write in Python with its tremendous stdlib. For example the tests I added re: rekeying -- awk would probably be a fine tool, but -- having written a modest amount of both at the hobbyist level -- I sure find it easier to have e.g. `first_word = output.strip().split()[0]` in my toolbox.